### ISSUE TITLE
New ESMF Errors from MAPL develop running Replay

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/MAPL#796]
My nightly tests have triggered new ESMF errors with MAPL `develop` when running replay. (In fact, it got flagged yesterday but there are currently so many "FAIL" in my nightly tests, that I'm missing things 😦 )

To wit, see the [attached ESMF log file](https://github.com/GEOS-ESM/MAPL/files/6279199/PET00.ESMF_LogFile.log) as there are a *LOT* of errors and warnings like:
```
20210408 024304.574 ERROR            PET00 ESMCI_Calendar.C:2015 ESMCI::Calendar::increment() Value cannot be a NULL pointer  - ; 'this' pointer or time argument is NULL.
20210408 024304.574 WARNING          PET00 ESMCI_Alarm.C:1301 ESMCI::Alarm::checkRingTime() Sticky alarm startReplay cannot be reversed since it has not been traversed forward and turned off via a user call to ESMF_AlarmRingerOff(), thereby enabling Time Manager to know the time to turn it back on in reverse.
20210408 024304.574 WARNING          PET00 ESMCI_Alarm.C:1301 ESMCI::Alarm::checkRingTime() Sticky alarm ReplayShutOff cannot be reversed since it has not been traversed forward and turned off via a user call to ESMF_AlarmRingerOff(), thereby enabling Time Manager to know the time to turn it back on in reverse.
20210408 024304.574 WARNING          PET00 ESMCI_Alarm.C:1301 ESMCI::Alarm::checkRingTime() Sticky alarm Time2Run cannot be reversed since it has not been traversed forward and turned off via a user call to ESMF_AlarmRingerOff(), thereby enabling Time Manager to know the time to turn it back on in reverse.
20210408 024304.574 WARNING          PET00 ESMCI_Alarm.C:1301 ESMCI::Alarm::checkRingTime() Sticky alarm PredictorAlarm cannot be reversed since it has not been traversed forward and turned off via a user call to ESMF_AlarmRingerOff(), thereby enabling Time Manager to know the time to turn it back on in reverse.
```

My *guess* is this is due to ESMF 8.1.0. As the GEOSgcm is now using that by default as of 2 days ago and I see no commits the last couple of days in MAPL concerning this.

The main person for this is probably @atrayano, but for now I'm adding @bena-nasa and @tclune as they are also much better at ESMF/MAPL than I am.

**Comment by mathomp4** [comment]
Note: Maybe this is part of the "GEOS needs updates to ESMF for alarms" thing that I've heard about in the CUPID meetings?

**Comment by atrayano** [comment]
@mathomp4 No, the errors and the warnings are real, and not related to ESMF 8.1.0. Only 1 of these messages comes from MAPL, related to record alarms, and lack of support from ESMF to check if a given alarm is already attached to the clock (sort of "if present"). The fix is ugly and tedious (get a full list of the alarms and scan it to find a match).  I will try to fix it soon. The rest of them can be fixed by small code changes in GEOS_GcmGridComp.F90, GEOS_AgcmGridComp.F90 and FV_StateMod.F90. These messages also exposed a very subtle bug in ESMF: if an alarm is sticky but disabled, regardless of how the clock is ticked (forward or reverse), this should not generate the warning message. But it does. And finally, I know why gfortran had issues with clock rewind some time ago - has to do with uninitialized time/calendar variables inside certain alarms - with the changes in gridded components (see above) this should be fixed.

**Comment by mathomp4** [comment]
@atrayano Has this been fixed? Or partially fixed? That is, are parts of the fixes in, say, MAPL, but waiting for fixes in the grid comps?

**Comment by rsdunlapiv** [comment]
@mathomp4 we have a branch of ESMF that we think will prevent that first ERROR you reported in the log related to a NULL pointer.

Could you please try the run again with this branch of ESMF?
https://github.com/esmf-org/esmf/tree/fixRingBegin

**Comment by mathomp4** [comment]
@rsdunlapiv Okay. I'll build and test it ASAP.

**Comment by mathomp4** [comment]
@rsdunlapiv The patch seems to prevent the error (running with multi on error)

**Comment by rsdunlapiv** [comment]
@mathomp4 Thanks for the update.  Thanks to @feiliuesmf for the fix.  When we have a new ESMF snapshot I will let you know.

**Comment by mathomp4** [comment]
@rsdunlapiv Thanks. I'll look at making a Baselibs for this soon.

**Comment by mathomp4** [comment]
Closed by updating to Baselibs 6.2.4
